File: Net40/Quicklogons.Client/Logic/LoginProvider.cs
using Quicklogons.Client.Data;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net;
using System.Runtime.Serialization.Json;
using System.Text;
using System.Threading.Tasks;
using System.Web;

namespace Quicklogons.Client.Logic
{
    public class LoginProvider : ILoginProvider
    {
        const string ReturnUrlQuery = "?returnurl={0}";
        const string RelativeLoginPageUrl = "/login/site/{0}";
        const string RelativeRequestUrl = "/login/site/{0}.json";

        ISettingsProvider _settingProvider;

        public LoginProvider(ISettingsProvider settingProvider)
        {
            _settingProvider = settingProvider;
        }

        public IEnumerable<ILoginLink> GetLoginLinks(string returnUrl = null)
        {
            var settings = _settingProvider.GetSettings();
            var relativeUrl = string.Format(RelativeRequestUrl, settings.SiteKey) + (!string.IsNullOrWhiteSpace(returnUrl) ? string.Format(ReturnUrlQuery, HttpUtility.UrlEncode(returnUrl)) : string.Empty);
            var requestUrl = new Uri(new Uri(settings.SecureUrl), relativeUrl);
            var rq = WebRequest.Create(requestUrl);
            rq.Method = "GET";
            var rs = rq.GetResponse();
            using (var stream = rs.GetResponseStream())
            {
                var js = new DataContractJsonSerializer(typeof(JsonLoginLink[]));
                return js.ReadObject(stream) as JsonLoginLink[];

            }
        }

        public string GetLoginPageLink(string returnUrl = null)
        {
            var settings = _settingProvider.GetSettings();
            var relativeUrl = string.Format(RelativeLoginPageUrl, settings.SiteKey) + (!string.IsNullOrWhiteSpace(returnUrl) ? string.Format(ReturnUrlQuery, HttpUtility.UrlEncode(returnUrl)) : string.Empty);
            var loginPageUri = new Uri(new Uri(settings.SecureUrl), relativeUrl);
            return loginPageUri.ToString();
        }
    }
}

File: php/Wordpress.plugin/quicklogons/logic/LoginProvider.php
<?php require_once("Settings.php"); ?>
<?php
class LoginProvider
{
    static $login_path = "/login/site/";

    private $settings;
    private $login_page_url;
        
    public static function create()
    {
        return new LoginProvider(Settings::load());
    }

    function __construct($_settings)
    {
        $this->settings = $_settings;
        $this->login_page_url = $this->settings->secure_url . LoginProvider::$login_path . $this->settings->site_key;
    }

    public function getLoginPageUrl($returnurl)
    {
        return isset($returnurl) && $returnurl > "" ? $this->login_page_url . "?returnurl=" . urlencode($returnurl) : $this->login_page_url;
    }

    public function getLoginLinks($returnurl)
    {
        $url = $this->settings->secure_url . LoginProvider::$login_path . $this->settings->site_key . ".json";
        $url = isset($returnurl) ? $url . "?returnurl=" . urldecode($returnurl) : $url;
        $json = file_get_contents($url);
        if ($json == FALSE) return FALSE;
        return json_decode($json);
    }
}
?>